### PR TITLE
Add optional arguments to startScan method and updated Readme

### PR DESCRIPTION
**Pull request by rohitsangwan01**

### CODE REVIEW

**Comment by CodeDoctorDE** [comment]
Why dynamic arguments?

**Comment by rohitsangwan01** [comment]
> Why dynamic arguments?

am working on Endorse web plugin , so that require a services list to pass in startScan method 
and many other advantages , like we can add multiple features to endorsed plugins , 
such as scan with services filter and timeout and many more

**Comment by Sunbreak** [comment]
I would prefer a standalone `requestDevice` API, i.e. https://github.com/woodemi/notepad_core/blob/b0e329f3d6e02f14f8a0e5e48a6ddb48e026b658/notepad_core_web/lib/notepad_core_web.dart#L33-L41

```dart
  @override
  Future<dynamic> requestDevice({
    List<String> optionalServices,
  }) async {
    return await bluetooth.requestDevice(ScanOptions(
      optionalServices: optionalServices.map(getServiceUUID).toList(),
      acceptAllDevices: true,
    ));
  }
```

File: quick_blue/pubspec.yaml [hunk 1]
```diff
@@ -17,6 +17,12 @@ dependencies:
   quick_blue_linux: ^0.1.2
   convert: ^3.0.0
 
+dependency_overrides:
```
[the remark this answers is not in the document]
Comment (Sunbreak) [reply quick_blue/pubspec.yaml:20]: There is `dependency_overrides` in quick_blue_example. No need in `quick_blue`

**Comment by fotiDim** [comment]
> I would prefer a standalone `requestDevice` API, i.e. https://github.com/woodemi/notepad_core/blob/b0e329f3d6e02f14f8a0e5e48a6ddb48e026b658/notepad_core_web/lib/notepad_core_web.dart#L33-L41
> 
> ```dart
>   @override
>   Future<dynamic> requestDevice({
>     List<String> optionalServices,
>   }) async {
>     return await bluetooth.requestDevice(ScanOptions(
>       optionalServices: optionalServices.map(getServiceUUID).toList(),
>       acceptAllDevices: true,
>     ));
>   }
> ```

@Sunbreak with `requestDevice()` how can we have unified API? If I understand correctly, the called (aka app developer) will have to call different APIs depending on the platform.

**Comment by Sunbreak** [comment]
Yep. Actually Web is quite different from others platforms. The way how API works and the user's interaction are quite different

**Pull request closed without merging**